File: demo/java/custom-java/src/main/java/com/baidu/bce/driver/Device.java
package com.baidu.bce.driver;

import com.baidu.bce.models.DeviceConfig;
import com.baidu.bce.sdk.context.models.device.DeviceInfo;

public class Device {
    private DeviceConfig cfg;
    private DeviceInfo info;
    private Simulator cli;

    public Device(DeviceConfig cfg, DeviceInfo info) {
        this.cfg = cfg;
        this.info = info;
        this.cli = new Simulator(cfg.getDevice());
    }

    public Object get(int index) throws Exception {
        return this.cli.get(index);
    }

    public void set(int index, float val) {
        this.cli.set(index, val);
    }

    public DeviceInfo getInfo() {
        return info;
    }

    public void setInfo(DeviceInfo info) {
        this.info = info;
    }
}

File: demo/csharp/custom-csharp/custom-csharp/Collection/Device.cs
using DriverSDK.DmContext.Models;
using Buildin.Csharp.Models;

namespace Buildin.Csharp.Collecton
{
    public class Device
    {
        private DeviceConfig cfg;
        private DeviceInfo info;
        private Simulator cli;

        public Device(DeviceConfig cfg, DeviceInfo info)
        {
            this.cfg = cfg;
            this.info = info;
            this.cli = new Simulator(cfg.Device);
        }

        public object Get(int index)
        {
            try
            {
                return this.cli.Get(index);
            }
            catch (Exception)
            {
                throw;
            }
        }

        public void Set(int index, double val)
        {
            this.cli.Set(index, val);
        }

        public DeviceInfo GetInfo()
        {
            return info;
        }

        public void SetInfo(DeviceInfo info)
        {
            this.info = info;
        }
    }
}

